frontend/pages/DashboardPage/cards/MDM/MDM.tsx

```diff
@@ -66,23 +72,44 @@ const Mdm = ({
   mdmSolutions,
   selectedPlatformLabelId,
   selectedTeamId,
+  onClickMdmSolution,
 }: IMdmCardProps): JSX.Element => {
   const [navTabIndex, setNavTabIndex] = useState(0);
 
   const onTabChange = (index: number) => {
     setNavTabIndex(index);
   };
 
+  const rolledupMdmSolutionsData = useMemo(() => {
+    if (!mdmSolutions) {
+      return [];
+    }
+
+    return mdmSolutions.reduce<IMdmSolution[]>((acc, nextSolution) => {
```

Comment:
To avoid the internal find on `acc`, what do you think about reducing to a dictionary first, then returning the array via `Object.values`?

```ts 
 return Object.values(mdmSolutions.reduce<Record<string, IMdmSolution>>((acc, nextSolution) => {
      if (acc[nextSolution.name])
       acc[nextSolution.name].hosts_count += nextSolution.hosts_count;
      } else {
        acc[nextSolution.name] = nextSolution;
      }
      return acc;
    }, {}));
```

Reply:
udpated